src/main/scala/uk/gov/nationalarchives/tdr/api/consignmentstatevalidation/ConsignmentStateTags.scala

```diff
@@ -0,0 +1,28 @@
+package uk.gov.nationalarchives.tdr.api.consignmentstatevalidation
+
+import java.util.UUID
+
+import sangria.execution.BeforeFieldResult
+import sangria.schema.{Argument, Context}
+import uk.gov.nationalarchives.tdr.api.graphql.{ConsignmentApiContext, ValidationTag}
+import uk.gov.nationalarchives.tdr.api.graphql.validation.UserOwnsConsignment
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait ConsignmentStateTag extends ValidationTag
+
+case class ValidateNoPreviousUploadForConsignment[T](argument: Argument[T]) extends ConsignmentStateTag {
```

Comment:
Sorry I didn't spot this before, but is this tag only going to be used on one endpoint? If so, you don't need the `arg match` statement below - you can just pull the argument out of the context like the simpler auth tags do:

```
val fileInput = ctx.arg[FileInputArg]("addFilesInput")
```

If we need to need to reuse this tag on another endpoint in future, we can think about commonising it then, because `UserOwnsConsignment` sounds quite auth-specific and might not be relevant here. (And if it _is_ relevant, we should rename it, or possibly think of a nicer way of wrapping the consignment ID than pattern matching every possible argument type.)

Reply:
Good point. Fixed